File: ios/StillMoment/Domain/Models/IntervalSettings.swift
//
//  IntervalSettings.swift
//  Still Moment
//
//  Domain Model - Interval Gong Settings for tick()
//

import Foundation

/// Configuration for interval gong detection during `MeditationTimer.tick()`.
///
/// Passed to `tick(intervalSettings:)` when interval gongs are enabled.
/// When `nil` is passed, no interval gong detection occurs.
struct IntervalSettings: Equatable {
    /// Interval in minutes between gongs (e.g., 5 for every 5 minutes)
    let intervalMinutes: Int

    /// The interval mode (repeating, afterStart, beforeEnd)
    let mode: IntervalMode
}

File: android/app/src/main/kotlin/com/stillmoment/domain/models/IntervalSettings.kt
package com.stillmoment.domain.models

/**
 * Configuration for interval gong detection during [MeditationTimer.tick].
 *
 * Passed to `tick(intervalSettings)` when interval gongs are enabled.
 * When `null` is passed, no interval gong detection occurs.
 *
 * @property intervalMinutes Interval in minutes between gongs (e.g., 5 for every 5 minutes)
 * @property mode The interval mode (REPEATING, AFTER_START, BEFORE_END)
 */
data class IntervalSettings(
    val intervalMinutes: Int,
    val mode: IntervalMode
)
